From 84b19e1f23d504b7830a96a2b9585c02ef98c072 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Sat, 10 Jun 2017 15:51:59 +0800
Subject: [PATCH 1885/5242] MLK-15030: EPDC: Filled stride value of s0_param
 layer

commit  152ab440331b95301ca8ec8d6205d84711ad937f from
https://source.codeaurora.org/external/imx/linux-imx.git

When epdc driver use pxp, it didn't fill stride value of s0_parm, so it leads to
epdc can display picture normally.

So fill it in this patch.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_epdc_v2_fb.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mxc_epdc_v2_fb.c b/drivers/video/fbdev/mxc/mxc_epdc_v2_fb.c
index 083b612..14bcb76 100644
--- a/drivers/video/fbdev/mxc/mxc_epdc_v2_fb.c
+++ b/drivers/video/fbdev/mxc/mxc_epdc_v2_fb.c
@@ -1828,6 +1828,7 @@ static int mxc_epdc_fb_set_par(struct fb_info *info)
 		}
 	}
 	pxp_conf->s0_param.width = screeninfo->xres_virtual;
+	pxp_conf->s0_param.stride = (screeninfo->bits_per_pixel * pxp_conf->s0_param.width) >> 3;
 	pxp_conf->s0_param.height = screeninfo->yres;
 	pxp_conf->s0_param.color_key = -1;
 	pxp_conf->s0_param.color_key_enable = false;
@@ -5503,6 +5504,7 @@ static int mxc_epdc_fb_probe(struct platform_device *pdev)
 	 */
 	pxp_conf->s0_param.pixel_fmt = PXP_PIX_FMT_RGB565;
 	pxp_conf->s0_param.width = fb_data->info.var.xres_virtual;
+	pxp_conf->s0_param.stride = (var_info->bits_per_pixel * pxp_conf->s0_param.width) >> 3;
 	pxp_conf->s0_param.height = fb_data->info.var.yres;
 	pxp_conf->s0_param.color_key = -1;
 	pxp_conf->s0_param.color_key_enable = false;
@@ -6031,6 +6033,7 @@ static int pxp_legacy_process(struct mxc_epdc_fb_data *fb_data,
 	struct dma_async_tx_descriptor *txd;
 	struct pxp_config_data *pxp_conf = &fb_data->pxp_conf;
 	struct pxp_proc_data *proc_data = &fb_data->pxp_conf.proc_data;
+	struct fb_var_screeninfo *screeninfo = &fb_data->info.var;
 	int i, ret;
 	int length;
 
@@ -6084,6 +6087,7 @@ static int pxp_legacy_process(struct mxc_epdc_fb_data *fb_data,
 	 * probe() and should not need to be changed.
 	 */
 	pxp_conf->s0_param.width = src_width;
+	pxp_conf->s0_param.stride = (screeninfo->bits_per_pixel * src_width) >> 3;
 	pxp_conf->s0_param.height = src_height;
 	proc_data->srect.top = update_region->top;
 	proc_data->srect.left = update_region->left;
-- 
1.7.9.5

